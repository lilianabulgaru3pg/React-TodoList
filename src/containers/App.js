import React, { Component, Suspense, lazy } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";

const LoginPage = lazy(() => import("./Login"));

class App extends Component {
  render() {
    return (
      <Router>
        <Suspense fallback={<div>Loading...</div>}>
          <Switch>
            <Route exact path="/" component={LoginPage} />
          </Switch>
        </Suspense>
      </Router>
    );
  }
}

export default App;
