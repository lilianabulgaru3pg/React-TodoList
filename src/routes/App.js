import React, { Suspense } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import RoutedFirewall from '../components/Firewall';
import { LOGIN } from '../constants';
import LoginPage from './Login/index';
import Tasks from './Tasks/index';

const App = () => (
  <Router>
    <Suspense fallback={<div>Loading...</div>}>
      <RoutedFirewall>
        <Tasks />
      </RoutedFirewall>
      <Route exact path={LOGIN} component={LoginPage} />
    </Suspense>
  </Router>
);

export default App;
