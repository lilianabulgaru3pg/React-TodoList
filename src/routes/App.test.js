import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import RoutedFirewall from '../components/Firewall';

describe('App', () => {
  const wrapper = shallow(<App />);

  it('should render a <BrowserRouter />', () => {
    expect(wrapper.find('BrowserRouter')).toHaveLength(1);
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render the Firewall', () => {
    expect(wrapper.find(RoutedFirewall).exists()).toBe(true);
  });
});
