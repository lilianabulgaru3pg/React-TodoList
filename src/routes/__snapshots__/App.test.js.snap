// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<BrowserRouter>
  <React.Suspense
    fallback={
      <div>
        Loading...
      </div>
    }
  >
    <withRouter(Firewall)>
      <Tasks />
    </withRouter(Firewall)>
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
  </React.Suspense>
</BrowserRouter>
`;
